dev/src/pool.ts

```diff
@@ -112,8 +118,12 @@ export class ClientPool<T> {
    */
   private async release(requestTag: string, client: T): Promise<void> {
     const requestCount = this.activeClients.get(client) || 0;
-    assert(requestCount > 0, 'No active request');
+    assert(
+      requestCount > 0,
+      'Release failed. No active requests are on the client.'
+    );
     this.activeClients.set(client, requestCount - 1);
+    this.operationsDeferred.resolve();
```

Comment:
`release()` gets called after every request. It does not mean that no clients have pending operations. If you want to keep `operationsDeferred`, I think you need logic here to determine whether there are any operations left.